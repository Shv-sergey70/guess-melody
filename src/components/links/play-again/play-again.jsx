import React from 'react';
import PropTypes from 'prop-types';
import {ActionCreator as GameActionCreator} from "../../../reducer/game/game";
import {ActionCreator as UserAnswersActionCreator} from '../../../reducer/user-answers/user-answers';
import {connect} from "react-redux";
import Simple from "../simple/simple";
import Route from '../../../routes';

const PlayAgain = ({onClick}) => {
  return (
    <Simple
      to={Route.MAIN}
      className="replay"
      onClick={onClick} >
      Сыграть ещё раз
    </Simple>
  );
};

PlayAgain.propTypes = {
  onClick: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onClick: () => {
    dispatch(GameActionCreator.replay());
    dispatch(UserAnswersActionCreator.resetAnswers());
  }
});

export {PlayAgain};
export default connect(null, mapDispatchToProps)(PlayAgain);
