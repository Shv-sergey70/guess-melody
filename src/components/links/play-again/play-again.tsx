import * as React from 'react';
import {ActionCreator as GameActionCreator} from "../../../reducer/game/game";
import {ActionCreator as UserAnswersActionCreator} from '../../../reducer/user-answers/user-answers';
import {connect} from "react-redux";
import Simple from "../simple/simple";
import Route from '../../../routes';

type Props = {
  onClick: ({}) => void
};

const PlayAgain: React.FunctionComponent<Props> = ({onClick}) => {
  return (
    <Simple
      to={Route.MAIN}
      className="replay"
      onClick={onClick} >
      Сыграть ещё раз
    </Simple>
  );
};

const mapDispatchToProps = (dispatch) => ({
  onClick: () => {
    dispatch(GameActionCreator.replay());
    dispatch(UserAnswersActionCreator.resetAnswers());
  }
});

export {PlayAgain};
export default connect(null, mapDispatchToProps)(PlayAgain);
