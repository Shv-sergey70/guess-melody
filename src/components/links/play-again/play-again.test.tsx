import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {PlayAgain} from "./play-again";

configure({adapter: new Adapter()});

test(`PlayAgainLink correctly renders`, () => {
  const welcomeScreenLink = shallow(<PlayAgain onClick={jest.fn()} />);

  expect(welcomeScreenLink).toMatchSnapshot();
});
