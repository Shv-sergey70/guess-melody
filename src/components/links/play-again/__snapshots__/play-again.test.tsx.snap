// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayAgainLink correctly renders 1`] = `ShallowWrapper {}`;
